eagles-food-backend/Controllers/AuthController.cs

```diff
@@ -38,13 +38,18 @@ public AuthController(IUserRepository userService)
         /// <param name="model">Request body with user's details</param>
         /// <returns>A response with 201 and the id, email of the new user</returns>
         /// <response code="201">Returns the id, email of the new user</response>
-        /// <response code="400">If the email is already taken, or invalid</response>
+        /// <response code="400">If the email is already taken, or phone number is invalid</response>
         /// <response code="500">If there was an error creating the user</response>
         [HttpPost("user/signup")]
         public async Task<ActionResult> CreateUser([FromBody] CreateUserDTO model)
         {
-            var res = await _userService.CreateUser(model);
-            return StatusCode((int)res.statusCode, res);
+            if(ModelState.IsValid)
+            {
+                var res = await _userService.CreateUser(model);
+                return StatusCode((int)res.statusCode, res);
+            }
+            
+            return StatusCode(400, $"{model.Phone} is not a valid number");
```

Comment:
is there any way to move the IsValid check into IUserRepository? so that maybe it could potentially be extended later on

Reply:
yhh there should be a way.
isValid put where it is atm acts as the first check bus stop before any action is processed.